cpp/include/cudf/reshape.hpp

```diff
@@ -97,6 +98,11 @@ std::unique_ptr<column> byte_cast(
   flip_endianness endian_configuration,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+std::pair<std::unique_ptr<column>, table_view> one_hot_encoding(
```

Comment:
Are there other names for this operation? Or would this be the most well known name for most people? It was not a name or an operation I was familiar with previously, but I'm not a data scientist :)

Reply:
Also, I'm unsure about `reshape` being the right place for this. `transform` perhaps, but `transform` is starting to become an island of misfit APIs. Correct me if I'm wrong, but this operation is more specific to feeding into ML kinds of algorithms, right? Perhaps there's a new header we need to create for this family of functionality.

Reply:
The (previously) used term for this in some literature is dummy encoding, but many people have made an effort to avoid this term as noted in https://developers.google.com/style/inclusive-documentation#ableist-language

The columns are essentially created by [indicator functions](https://en.wikipedia.org/wiki/Indicator_function) and sometimes called [indicator variables](https://online.stat.psu.edu/stat462/node/161/) in stats, which could perhaps be considered here.

Reply:
So after my very thorough 30 seconds of Googling, it appears that this is a feature where both Pandas and sklearn have solutions. Pandas has `get_dummies` and sklearn has `OneHotEncoder`. I don't have any intuition or context for understanding if equivalent functionality makes more sense to live in cuDF, cuML, or both. Do you have any thoughts on that @beckernick?

Reply:
I believe currently cudf has a `one_hot_encoding` (supporting get_dummies) [feature](https://github.com/rapidsai/cudf/blob/cc1306016e4d188dd8d4158b0edafa4c68fbeed9/python/cudf/cudf/core/series.py#L3487). Per #8870, python overhead is too high when cardinality is large. So this PR is meant to evade them via a libcudf implementation.

Reply:
Okay, so I guess that answers that question.

Reply:
As @beckernick pointed out, the use of the term "dummy" is something that we're trying to avoid, but for context the use of "dummy variable" vs "one-hot encoding" is mostly a difference between fields. Dummy variables are historically used by statisticians. I believe the most common usage of the term is in building up regression models etc where a categorical variable needs to be represented within a continuous regression fit, very similar to the use of mixed-integer linear programming to solve certain classes of optimization problems. One-hot is the common term in machine learning, and it's commonly used for things like neural networks where you need to transform between continuous values and categoricals i.e. a classification problem. This difference is reflected in the fact that you found `pandas.get_dummies` vs `sklearn.preprocessing.OneHotEncoder`. For obvious reasons the "one-hot" terminology has become pretty common in the last few years, but I don't think it was used much before ML became so popular (maybe in circuits a bit).

I think cudf is a fine place for this to live, cuml could always call it directly if they want to also alias that sklearn functionality. I agree that `reshape` may not be the best place for this API though. `transform` also feels a bit odd, but I agree that it's probably the best that we have right now. The output is basically a different encoding of the same data, so I guess it's sort of similar to `encode` :shrug:

Reply:
Updated under `transform.hpp`, next to encode. Until we find a better home for it.